src/sets.jl

```diff
@@ -80,6 +82,40 @@ function dual_set_type end
 
 dual_set_type(S::Type{<:AbstractSet}) = error("Dual type of $S is not implemented.")
 
+
+"""
+    AbstractDistance
+
+Distance function used to evaluate the distance from a point to a set.
+New subtypes of `AbstractDistance` must implement fallbacks for sets they don't cover and implement
+`distance_to_set(::Distance, v, s::S)` for sets they override the distance for.
```

Comment:
> This is still an issue.
 
mistake on my part this should be removed

Reply:
> I thought we were providing the AbstractDistance fallback?

we can provide one d(::AbstractDistance, v, s) = d(::DefaultDistance, v, s) but not be more specific because of method ambiguity